### PR TITLE
[ZEPPELIN-1346] Add contribution guidelines to the website

### PR DESCRIPTION
**Pull request by corneadoug**
### What is this PR for?
This PR is bringing Contribution guidelines to the website.
We also introduce a sidemenu on those pages allowing to create additional content.
As an example, for the Web Application Guidelines, we added a page about "Defining Components"

In a different PR, we will remove the contributing markdown, and change the links in the repository to the website.

### What type of PR is it?
Documentation

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1346

### How should this be tested?
Build the website locally and visit the community page

### Screenshots (if appropriate)
screen shot 2016-08-23 at 3 50 41 pm

screen shot 2016-08-23 at 3 50 51 pm

screen shot 2016-08-23 at 3 50 33 pm

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: contribution/general.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+* Anybody can be a reviewer and may comment on the change and suggest modifications.
+* Reviewer can indicate that a patch looks suitable for merging with a comment such as: "Looks good", "LGTM", "+1".
+* At least one indication of suitable for merging (e.g. "LGTM") from committer is required to be merged.
+* Pull request is open for 1 or 2 days for potential additional review, unless it's got enough indication of suitable for merging.
+* Committer can initiate lazy consensus ("Merge if there is no more discussion") and the code can be merged after certain time (normally 24 hours) when there is no review exists.
+* Contributor can ping reviewers (including committer) by commenting 'Ready to review' or suitable indication.
+
+## Becoming a Committer
+
+The PPMC adds new committers from the active contributors, based on their contribution to Zeppelin. The qualifications for new committers include:
```
Comment (bzz) [inline contribution/general.md]: PPMC -> PMC

File: contribution/general.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+## Tests
+Each new File should have its own accompanying unit tests. Each new interpreter should have come with its tests.
+
+
+Zeppelin has 3 types of tests:
+
+  1. Unit Tests: The unit tests run as part of each package's build. E.g. SparkInterpeter Module's unit test is SparkInterpreterTest
+  2. Integration Tests: The integration tests run after all modules are build. The integration tests launch an instance of Zeppelin server. ZeppelinRestApiTest is an example integration test.
+  3. GUI integration tests: These tests validate the Zeppelin UI elements. These tests require a running Zeppelin server and launches a web browser to validate Notebook UI elements like Notes and their execution. See ZeppelinIT as an example.
```
Comment (bzz) [inline contribution/general.md]: May be it's a good time to add here a CLI command to run Integration Tests?

File: contribution/zeppelinweb/GPG01.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+---
+layout: sideMenu
```
Comment (bzz) [inline contribution/zeppelinweb/GPG01.md]: File name is a bit confusing here as it's not related to GPG aka GnuPG https://www.gnupg.org/

File: contribution/general.md (hunk 2 above)
Comment (corneadoug) [inline contribution/general.md]: If you have the CLI command actually that would be nice

**Comment by bzz** [comment]
Overall looks great to me, thank you @corneadoug for taking care! Few minor comments above.
\cc @AhyoungRyu for review

File: _includes/sub-views/community/contribute.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+## How to contribute
+
+There is multiple ways you can contribute to the project.
```
Comment (AhyoungRyu) [inline _includes/sub-views/community/contribute.md]: There are~??

File: _includes/sub-views/community/contribute.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+And help is always welcome!
+
+#### Issue Tracker
```
Comment (AhyoungRyu) [inline _includes/sub-views/community/contribute.md]: it's a nitpick, can we change this header tag to `###`? `####` font size is not much different with a plain text ㅜ_ㅜ

The other titles too :)

File: _includes/sub-views/community/contribute.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+Don't hesitate to report new bugs and improvements ideas.
+
+#### Contribute to the source code
+
+We like to get code contributions through Pull Requests on our [Github Mirror](https://github.com/apache/zeppelin).
+
+But before starting, please read our [Contribution guidelines](/contribution/general.html), it will give
+you important information about our review process, and pointers on how to make a good code contribution.
+
+You can visit our [issue Tracker](https://issues.apache.org/jira/browse/ZEPPELIN) to find issues to resolve,
```
Comment (AhyoungRyu) [inline _includes/sub-views/community/contribute.md]: `issue Tracker` -> `Issue Tracker` (Change to capital `I`) :D

File: _includes/sub-views/community/contribute.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+and if your are a newcomer and don't know where to get started, we have set some [beginner issues](https://issues.apache.org/jira/browse/ZEPPELIN-1245?jql=project%20%3D%20ZEPPELIN%20AND%20status%20%3D%20Open%20AND%20labels%20%3D%20beginner).
+
+#### Other contributions
+
+Not much of a coder? There is other ways to help out:
```
Comment (AhyoungRyu) [inline _includes/sub-views/community/contribute.md]: There "are" other ways

**Comment by AhyoungRyu** [comment]
@corneadoug  Great work! Definitely this way would be better than we put `CONTRIBUTING.md` under each component dir in Zeppelin github. I also left some minor comments. 
One question. If this PR can cover the other contributing guide(not only `web application`), can we put `docs/CONTRIBUTING.md` as well? What do you think? I can create a PR to your repo for that if you're positive :D

**Comment by corneadoug** [comment]
@AhyoungRyu That's the goal of having the side menu, to be able to have contribution guide focused on specific components (Back-end, front-end, interpreters, website, docs etc..)

I only wanted to cover the structure in this PR, that's why there is web application as example.
So you can either do a PR after this one is merged, or directly in this PR, your choice!

**Comment by AhyoungRyu** [comment]
@corneadoug Thank you for the explanation. Since it's a only one page, i've created a PR at your repo :) 
(Just combined `docs/README.md` & `docs/CONTRIBUTING.md` contents)

**Comment by corneadoug** [comment]
@bzz I added a section with the Selenium CLI command, can you double check it?

File: contribution/zeppelinweb/goodPracticeGuide01.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  function HomeCtrl($location) {
+    ...
+  }
+
+})();
+```
+
+#### Explanations
+
+* The component function and the component's dependency injection is separated from the component definition
```
Comment (AhyoungRyu) [inline contribution/zeppelinweb/goodPracticeGuide01.md]: @corneadoug Sorry again. I think `is` -> `are separated` is right :)

**Comment by corneadoug** [comment]
All the changes have been addressed. LGTM?

**Comment by AhyoungRyu** [comment]
LGTM 👍

**Pull request closed without merging**